{
  AssertableLogProvider provider=new AssertableLogProvider();
  ErrorReporter reporter=new ErrorReporter(provider.getLog("userlog"),new UsageData());
  Throwable cause=new Throwable("This is not an error we know how to handle.");
  Neo4jError error=Neo4jError.from(cause);
  reporter.report(error);
  assertThat(error.status(),equalTo((Status)Status.General.UnknownFailure));
  provider.assertExactly(inLog("userlog").error(both(containsString("START OF REPORT")).and(containsString("END OF REPORT"))));
  provider.assertExactly(inLog("userlog").error(containsString(error.reference().toString())));
}
